docs/docs/security-considerations.md

```diff
@@ -0,0 +1,54 @@
+# Security Considerations
+
+Contrast ensures application integrity and provides secure means of communication and bootstrapping (see [security benefits](basics/security-benefits.md)).
+However, care must be taken when interacting with the outside of Contrast's confidential environment.
+This page presents some tips for writing secure applications and outlines the trust boundaries app developers need to know.
+
+## General recommendations
+
+### Authentication
+
+The application receives credentials from the Contrast Coordinator during initialization.
+This allows to authenticate towards peers, and to verify credentials received from peers.
+The application should use the certificate bundle to authenticate incoming requests, and be wary of unauthenticated requests or requests with a different root of trust (e.g. internet PKI).
+
+This recommendation applies to network traffic, but also to volumes, GPUs and other devices.
+Generally speaking, all information provided by the world outside the confidential VM should be treated with due scepticism, especially if it is not authenticated.
+Examples include dynamic information from the k8s API server, the kubelet, DNS and others.
+
+### Encryption
+
+Any external persistence should be encrypted with an authenticated cipher.
+This recommendation applies to block devices or filesystems mounted into the container, but also to cloud blob storage or external databases.
+
+## Contrast security guarantees
+
+If an application authenticates with a certificate signed by the Contrast Mesh CA of a given manifest, Contrast provides the following guarantees:
+
+1. The container images used by the app are the images specified in the resource definitions.
+2. The command line arguments of containers are exactly the arguments specified in the resource definitions.
+3. All environment variables are either specified in resource definitions, in the container image manifest or in a settings file for the Contrast CLI.
+4. The containers run in a confidential VM that matches the reference values in the manifest.
+5. The containers' root filesystems are mounted in encrypted memory.
+
+### Limitations inherent to policy checking
+
+Workload policies serve as workload identities.
+From the perspective of the Contrast Coordinator, all workloads that authenticate with the same policy are equal.
+Thus, it's not possible to disambiguate, for example, pods spawned from a deployment, or to limit the amount of certificates issued per policy.
+
+Container image references from Kubernetes resource definitions are taken into account when generating the policy.
+A mutable reference may lead to policy failures or unverified image content, depending on the Contrast runtime.
```

Comment:
I know that it's quite clear, but an example, like "docker.io/nginx:latest" or a link to some explaining resource could help.